cmake/hwmalloc.cmake

```diff
@@ -25,6 +25,10 @@ set(_hwmalloc_tag        "master")
 if(NOT _hwmalloc_already_fetched)
     message(STATUS "Fetching HWMALLOC ${_hwmalloc_tag} from ${_hwmalloc_repository}")
 endif()
+
+if(EXISTS ${PROJECT_SOURCE_DIR}/extern/hwmalloc)
+    set(FETCHCONTENT_SOURCE_DIR_HWMALLOC ${PROJECT_SOURCE_DIR}/extern/hwmalloc)
+endif()
```

Comment:
this makes Cmake look at this directory, do I understand this correctly? Would it make sense to make hwmalloc a git-submodule instead? I agree that the hardwired fetch from the repo is bad practice.

Reply:
Having worked with many projects that use fetchcontent, it is often the case, that you are making changes to the main project, as well as the subproject and you want the subproject to be in a location other than the default one set by cmake - which is in the build dir - and gets wiped regularly if you clean that dir to fix some cmake cache type of issue. Allowing the sub-project to 'live' in the source dir and be picked up automatically is convenient.

I do not think it is necessary to use git submodules, fetchcontent does what's needed, and allowing a check for a copy in some user defined place (cmake var can be used instead of this hard check, but it's just a conenience) - allows the best of both worlds.

Reply:
ok